crates/sui-types/src/messages.rs

```diff
@@ -833,6 +833,11 @@ pub enum ObjectInfoRequestKind {
     LatestObjectInfo(Option<ObjectFormatOptions>),
     /// Request the object state at a specific version
     PastObjectInfo(SequenceNumber),
+    /// Similar to PastObjectInfo, except that it will also return the object content.
+    /// This is used only for debugging purpose and will not work in the long run when
+    /// we stop storing all historic versions of every object.
+    /// No production code should depend on this kind.
+    PastObjectInfoDebug(SequenceNumber, Option<ObjectFormatOptions>),
```

Comment:
another option would be to have PastObjectInfo return an Option, so it can return the object if it still exists. this would make perfect sense to have on an archival node or sparse node where we might not prune old versions.

Reply:
This API is specifically for validator and not really sharable. If in the future we need this on archival node we will probably be providing a separate API